### DISCUSSION TITLE
ghcr.io/itzg/ Minecrem-server :java21

### DISCUSSION THREAD

**Discussion by Amatsuka-Mao** [itzg/docker-minecraft-server#2952] [Q&A]
### Describe the problem

I started the container after using this command docker pull ghcr.io/itzg/ Minecrem-server :java21 and the log showed me ERROR: your Docker provider has an annoying flaw where it  tries to set PATH even though the container establishes  a very  specific value Why is that?

### Container definition

```
Paste run command or compose file here
```

### Container logs

```
Paste logs here
```
[init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 4 1000 1000 4096 Jun 18 09:43 /data'

[init] ERROR: your Docker provider has an annoying flaw where it

[init]        tries to set PATH even though the container establishes

[init]        a very specific value.

[init] ERROR: could not locate path that contains java

**Comment by Amatsuka-Mao** [comment]
I use the following command to run the container docker run -d -v /path/on/host:/data -e TYPE=FABRIC -p 25565:25565-e EULA=TRUE --name mc ghcr.io/itzg/minecraft-server:java21-jdk

**Comment by itzg** [comment]
I am unable to recreate this issue. Please provide the output the following commands:

```
docker image inspect -f '{{json .Config.Labels}}' ghcr.io/itzg/minecraft-server:java21-jdk
docker run --rm --entrypoint bash ghcr.io/itzg/minecraft-server:java21-jdk -c 'ls -l /opt/java/openjdk/bin'
```

**Comment by Amatsuka-Mao** [comment]
> 我无法重现此问题。请提供以下命令的输出：
> 
> ```
> docker image inspect -f '{{json .Config.Labels}}' ghcr.io/itzg/minecraft-server:java21-jdk
> docker run --rm --entrypoint bash ghcr.io/itzg/minecraft-server:java21-jdk -c 'ls -l /opt/java/openjdk/bin'
> ```

{“org.opencontainers.image.authors”：“Geoff Bourne [itzgeoff@gmail.com](mailto:itzgeoff@gmail.com)”，“org.opencontainers.image.created”：“2024-06-15T19：05：30.508Z”，“org.opencontainers.image.description”：“提供Minecraft服务器的Docker映像，该镜像将在启动时自动下载所选版本”，“org.opencontainers.image.licenses”：“Apache-2.0”，“org.opencontainers.image.ref.name”：“ubuntu”，“org.opencontainers.image.revision”：“fe5f066a0be404d0bae91fc8fa928bb918c22bc2”，”org.opencontainers.image.source“：”[https://github.com/itzg/docker-minecraft-server“，”org.opencontainers.image.title“：”docker-minecraft-server“，”org.opencontainers.image.url“：”https://github.com/itzg/docker-minecraft-server“，”org.opencontainers.image.version“：”java21-jdk“}](https://github.com/itzg/docker-minecraft-server%22,%22org.opencontainers.image.title%22:%22docker-minecraft-server%22,%22org.opencontainers.image.url%22:%22https://github.com/itzg/docker-minecraft-server%22,%22org.opencontainers.image.version%22:%22java21-jdk%22%7D)

总计 460
-rwxr-xr-x 1 根根 12304 Apr 17 07：：05 jar
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jarsigner
-rwxr-xr-x 1 根根 12248 4 月 17 日 07：05 java
-rwxr-xr-x 1 根根 12336 4 月 17 日 07：05 javac
-rwxr-xr-x 1 根根 12336 4 月 17 日 07：05 javadoc
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 javap
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jcmd
-rwxr-xr-x 1 根根 12368 4 月 17 日 07：05 jconsole
-rwxr-xr-x 1 根根 12304 Apr 17 07：05 jdb
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jdeprscan
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jdeps
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jfr
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jhsdb
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jimage
-rwxr-xr-x 1 根根 12336 4 月 17 日 07：05 jlink

-rwxr-xr-x 1 根根 12336 4 月 17 日 07： 05 jmap
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jmod
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jpackage
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jps
-rwxr-xr-x 1 根根 12344 4 月 17 日 07：05 jrunscript
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jshell
-rwxr-xr-x 1 根根 12336 4 月 17 日 07：05 jstack
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jstat
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 jstatd
-rwxr-xr-x 1 根根 12312 4 月 17 日 07：05 jwebserver
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 keytool
-rwxr-xr-x 1 根根 12336 4 月 17 日 07：05 rmiregistry
-rwxr-xr-x 1 根根 12304 4 月 17 日 07：05 serialver

**Comment by Amatsuka-Mao** [comment]
> 我无法重现此问题。请提供以下命令的输出：
> 
> ```
> docker image inspect -f '{{json .Config.Labels}}' ghcr.io/itzg/minecraft-server:java21-jdk
> docker run --rm --entrypoint bash ghcr.io/itzg/minecraft-server:java21-jdk -c 'ls -l /opt/java/openjdk/bin'
> ```

The latest package you released has solved my problem ,Thank you .
